emcc.py

```diff
@@ -1242,6 +1242,12 @@ def check(input_file):
       if shared.Settings.WASM:
         shared.Settings.BINARYEN = 1 # these are synonyms
 
+        # When only targeting wasm, the .asm.js file is not executable, so is treated as an intermediate build file that can be cleaned up.
+        if shared.Building.is_wasm_only():
+          asm_target = asm_target.replace('.asm.js', '.temp.asm.js')
```

Comment:
do we even need the rename? i think this line can be removed

lgtm with that, and with testing. one of the `other.test_binaryen*` tests might already check for build outputs so it could be added to (or if not, please add a new test)